src/nni_manager/training_service/reusable/environment.ts

```diff
@@ -125,8 +125,9 @@ export abstract class EnvironmentService {
     public abstract get hasStorageService(): boolean;
     public abstract config(key: string, value: string): Promise<void>;
     public abstract refreshEnvironmentsStatus(environments: EnvironmentInformation[]): Promise<void>;
-    public abstract startEnvironment(environment: EnvironmentInformation): Promise<void>;
     public abstract stopEnvironment(environment: EnvironmentInformation): Promise<void>;
+    public abstract startEnvironment(environment: EnvironmentInformation): Promise<void>;
+    public abstract getInitializeEnvironmentNumber(): number;
```

Comment:
1. How about merge this function with hasStorageService, environmentMaintenceLoopInterval? they can return a structure.
2. getInitializeEnvironmentNumber is a weired name, how about prefetchedEnvironmentCount?
3. refer to hasMoreEnvironments to provide a default implementation, and return 0. So that other service don't need to repeat to return 0.

Reply:
changed to use prefetchedEnvironmentCount(). I think it is better split these three data, for different platform has different data type and value. for example, openPAI only need to set hasStorageService, while remote need to set prefetchedEnvironmentCount, split them is more flexible.